radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -91,31 +92,33 @@ public PendingVotes(Hasher hasher) {
 	 * @return The generated QC, if any
 	 */
 	public Optional<QuorumCertificate> insertVote(Vote vote, BFTValidatorSet validatorSet) {
-		final VoteData voteData = vote.getVoteData();
-		final Hash voteHash = this.hasher.hash(voteData);
-		final ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
 		final BFTNode node = vote.getAuthor();
 		// Only process for valid validators and signatures
 		if (!validatorSet.containsNode(node)) {
 			log.info("Ignoring vote from invalid author {}", node::getSimpleName);
 			return Optional.empty();
 		}
 
+		final TimestampedVoteData timestampedVoteData = vote.getTimestampedVoteData();
+		final VoteData voteData = timestampedVoteData.getVoteData();
+		final Hash voteDataHash = this.hasher.hash(voteData);
 		final View voteView = voteData.getProposed().getView();
-		if (!replacePreviousVote(node, voteView, voteHash)) {
+		if (!replacePreviousVote(node, voteView, voteDataHash)) {

```

Comment:
Would be good to write integrationTest with a leader sending out different QCs

Reply:
OK, done. Test is in `DifferentTimestampsCauseTimeoutTest`.